Setup script

mcgill-a opened the pull request:
I have created a bash script to automatically setup and run the python virtual environment and web server.

It also generates the required "config/defaults.py" configuration file if it is doesn't exist. (The configuration file is added to a .gitignore to prevent leaking Spotify developer API keys).

![image](https://user-images.githubusercontent.com/31523920/108999683-8d253880-769a-11eb-807e-96820b93eb8b.png)
![image](https://user-images.githubusercontent.com/31523920/108999664-84346700-769a-11eb-927c-f3e0efea04f7.png)

mcgill-a committed:
bash setup script

mcgill-a committed:
Improved setup script compatibility

mcgill-a committed:
More checks to see if prerequisites are already setup

mcgill-a committed:
Detect invalid configuration during setup

mcgill-a committed:
Merge branch 'setup-script' of https://github.com/mcgill-a/whichone into setup-script

mcgill-a committed:
Added error message for missing python

athompsonScottLogic reviewed (approved):
Great this makes the process a lot easier

connor-settle commented on setup.sh:
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
I guess its fine to have a bash setup script. 

Normally I'd expect solutions to have a readme with prerequisites section (with relevant links) and Getting Started section. A good place to explain the purpose of command line args if you decide to make use of them (for credentials/settings)

connor-settle replied:
If you agree about having README.md and command line args rather than bash setup and a config file I'd suggest its out of scope for this PR. You can add a task in trello for that.

mcgill-a replied:
Okay yeah that's a fair point, I'll add some more detailed instructions to the existing readme and look into using command line arguments instead

connor-settle commented on whichone/__init__.py:
```diff
@@ -25,6 +25,12 @@
+    elif (app.config['SPOTIFY_CLIENT_ID'] == "YOUR SPOTIFY DEV CLIENT ID HERE"):
+        print("Insert a valid SPOTIFY_CLIENT_ID in /config/defaults.py")
+        exit(0)
+    elif (app.config['SPOTIFY_CLIENT_SECRET'] == "YOUR SPOTIFY DEV CLIENT SECRET'"):
```
Can these be passed as command line args?

mcgill-a replied:
Currently they're only being read trough a config file but that's a good idea. Would you say remove them from the config entirely and just pass them in as command line arguments every time?

mcgill-a committed:
Implemented command line arguments instead of configuration file

mcgill-a committed:
removed setup.sh

mcgill-a commented:
In the latest commit I have removed setup.sh and implemented command line arguments for all configuration parameters instead of the defaults.py file. This means the user can pass in parameters at run time and set default environment variables if they want to pass the same ones in every time without typing them out. For example:
![image](https://user-images.githubusercontent.com/31523920/109035968-0c2d6780-76c1-11eb-8c9e-f607885948a3.png)
![image](https://user-images.githubusercontent.com/31523920/109034974-14d16e00-76c0-11eb-82a4-855ff0b96043.png)

Also, the LIMIT_API and LIMIT_ROUTE parameters are optional parameters because they both have a default value. This means the only parameters that need to be entered are:
`python run.py -id $SPOTIFY_CLIENT_ID -secret $SPOTIFY_CLIENT_SECRET -redirect $SPOTIFY_REDIRECT_URI`

connor-settle reviewed (approved):
Looks great.

mcgill-a committed:
switched argument input key to '--' instead of '-'
